x/programs/rust/examples/single_nft/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+const TOTAL_SUPPLY: i64 = 1;
+
+/// The program storage keys.
+#[state_keys]
+enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
+    /// The name of the token. Key prefix 0x1.
+    Name,
+    /// The symbol of the token. Key prefix 0x2.
+    Symbol,
+    /// Metadata of the token. Key prefix 0x3.
+    Uri,
+    /// Balance of the NFT token by address. Key prefix 0x4(address).
+    Balance(Address),
+    /// Counter -- used to keep track of total NFTs minted. Key prefix 0x5.
+    Counter,
+    /// Owner -- used to keep track of the owner. Key prefix 0x6.
+    Owner,
+}
+
+/// Initializes the NFT with all required metadata.
+/// This includes the name, symbol, image URI, owner, and total supply.
+/// Returns true if the initialization was successful.
+#[public]
+pub fn init(
+    program: Program,
+    name_ptr: i64,
+    nft_name_length: i64,
+    nft_symbol_ptr: i64,
+    nft_symbol_length: i64,
+    nft_uri_ptr: i64,
+    nft_uri_length: i64,
+) -> bool {
+    let name_ptr = Memory::new(Pointer::from(name_ptr));
+    let nft_name = unsafe { name_ptr.range(nft_name_length as usize) };
+
+    let nft_symbol_ptr = Memory::new(Pointer::from(nft_symbol_ptr));
+    let nft_symbol = unsafe { nft_symbol_ptr.range(nft_symbol_length as usize) };
+
+    let nft_uri_ptr = Memory::new(Pointer::from(nft_uri_ptr));
+    let nft_uri = unsafe { nft_uri_ptr.range(nft_uri_length as usize) };
+
+    // Set token name
+    program
+        .state()
+        .store(StateKey::Name.to_vec(), &nft_name)
+        .expect("failed to store nft name");
+
+    // Set token symbol
+    program
+        .state()
+        .store(StateKey::Symbol.to_vec(), &nft_symbol)
+        .expect("failed to store nft symbol");
+
+    // Set token URI
+    program
+        .state()
+        .store(StateKey::Uri.to_vec(), &nft_uri)
+        .expect("failed to store nft uri");
+
+    // Set total supply
+    program
+        .state()
+        .store(StateKey::TotalSupply.to_vec(), &TOTAL_SUPPLY)
+        .expect("failed to store total supply");
+
+    // Initialize counter
+    program
+        .state()
+        .store(StateKey::Counter.to_vec(), &0_i64)
+        .expect("failed to store counter");
+
+    true
+}
+
+/// Mints NFT tokens and sends them to the recipient.
+#[public]
+pub fn mint(program: Program, recipient: Address) -> bool {
+    const MINT_AMOUNT: i64 = 1;
+
+    let counter = program
+        .state()
+        .get::<i64, _>(StateKey::Counter.to_vec())
+        .expect("failed to store ");
+
+    assert!(counter <= TOTAL_SUPPLY, "max supply for nft exceeded");
+
+    let balance = program
+        .state()
+        .get::<i64, _>(StateKey::Balance(recipient).to_vec())
+        .unwrap_or_default();
+
+    program
+        .state()
+        .store(
+            StateKey::Balance(recipient).to_vec(),
+            &(balance + MINT_AMOUNT),
+        )
+        .expect("failed to store balance");
+
+    program
```

Comment:
It seems odd to update the owner to whoever performed the last mint?

Reply:
I don't believe you are associating this with the NFT ID minted (maybe what you intended)?